### PR TITLE
Update the deployment procedure

### PR DESCRIPTION
**Pull request by Garanas**
## Description of the proposed changes

Update the deployment procedure together with @BlackYps . The majority of the steps should now be described.

For the other half of the changes, see also: https://github.com/FAForever/fa/commit/a3de04de69292a401a0dd646397e62e18a779f8b

### CODE REVIEW

**Comment by Garanas** [comment]
After discussing it with @Sheikah45 , the deployment of the executable is as simple as copying whatever is the executable for FAF Develop is to the folder for FAF on the server. That means updating the executable for FAF Develop is **necessary** before asking the server administrators to update the executable.

File: docs/deployment.md [hunk 1]
```diff
@@ -51,28 +55,35 @@ permalink: patches/3811
...
-- (6) Stage, commit and push the changes to GitHub. Create a pull request on GitHub to allow other maintainers to review the changelog. Make sure the pull requests points to `develop`. 
+- (6) Stage, commit and push the changes to GitHub. Create a pull request on GitHub to allow other maintainers to review the changelog. Make sure the pull requests points to `develop`.
 - (7) Delete the current snippets and stage, commit and push the changes to GitHub.
 
 You can re-use the same branch and pull request in the next phase of the deployment.
 
 ### Deployment of Lua
 
-The following (manual) steps are relevant to create a valid deployment to the FAF game type.
+- (1) Checkout on the `changelog branch` branch created in the previous step and pull in the latest version.
```
Comment (BlackYps) [inline docs/deployment.md:65]: These lines are kind of redundant now. I would just get rid of them, turning these two sections into one.

File: docs/deployment.md [hunk 2]
```diff
@@ -21,14 +21,18 @@ The deployment procedure can be a lengthy process because it involves various st
 
 ### Deployment of engine patches
 
-The deployment of engine patches to the release branch is a manual process. This is intentional as users will automatically download the executable upon launching the game. 
+The deployment of engine patches to the release branch is a manual process. This is intentional as users will automatically download the executable upon launching the game.
```
[the remark this answers is not in the document]
Comment (Garanas) [reply docs/deployment.md]: Processed in: https://github.com/FAForever/fa/pull/6409/commits/1060966192268e426bd0cf1d45310f7f88492de8

File: docs/deployment.md [hunk 3]
```diff
@@ -21,14 +21,18 @@ The deployment procedure can be a lengthy process because it involves various st
...
 - (2) Update the executable of the `FAF Develop` and `FAF Beta Balance` game types using the [Upload workflow](https://github.com/FAForever/FA-Binary-Patches/actions).
-- (3) Ask a server administrator to prepare the executable to be updated upon the next game release.
-- (4) ???
 
-You can continue the deployment steps but you can not finalize it until the server administrator got back to you that it is set. This may take an arbitrary amount of time so make sure this is done well in advance. 
+The workflow requires an approval of another maintainer. Once approved, wait for the workflow to finish.
+
+- (3) Verify the executable on the `FAF Develop` game type.
+- (4) Ask a server administrator to prepare the executable to be updated upon the next game release.
+- (5) ???
```
[the remark this answers is not in the document]
Comment (Garanas) [reply docs/deployment.md]: It's missing step (6): profit!

But in reality, I had to ask Sheikah what the process is exactly. Processed in: https://github.com/FAForever/fa/pull/6409/commits/c1462614b8f2517590eb250ad692243cf204d018

File: docs/deployment.md (hunk 1 above)
Comment (Garanas) [reply docs/deployment.md:65]: The Lua changes are separate from the changelog thematically. Can you make a suggestion on how you'd want this part of the documentation to be?

Comment (BlackYps) [reply docs/deployment.md:65]: Sure

Comment (BlackYps) [reply docs/deployment.md:65]: Ah, github doesn't let me create a suggestion here, so I'll have to do it more manually.
Replace all this
```
You can re-use the same branch and pull request in the next phase of the deployment.

### Deployment of Lua

- (1) Checkout on the `changelog branch` branch created in the previous step and pull in the latest version.
- (2) Update the game version in [mod_info.lua](https://github.com/FAForever/fa/blob/develop/mod_info.lua) and [version.lua](https://github.com/FAForever/fa/blob/develop/lua/version.lua).
```
with 
```
- (8) Update the game version in [mod_info.lua](https://github.com/FAForever/fa/blob/develop/mod_info.lua) and [version.lua](https://github.com/FAForever/fa/blob/develop/lua/version.lua) and stage, commit and push the changes to GitHub.
```

**Comment by BlackYps** [comment]
rerequest my review when you are ready

File: docs/deployment.md [hunk 4]
```diff
@@ -44,35 +48,42 @@ You can continue the deployment steps but you can not finalize it until the serv
...
+- - (1.1) Set the tag with the game version. 
+- - (1.2) Match the format of the title with that of previous releases.
+- - (1.3) Copy and paste the changelog into the description. Make sure to remove the title as a release has its own title.
+- - (1.4) Create the release.
+- (2) Use the [Deploy to FAF Workflow](https://github.com/FAForever/fa/actions/workflows/deploy-faf.yaml) to perform the deployment.
 
+The workflow requires an approval of another maintainer. Once approved, wait for the workflow to finish.
 
+- (3) Use the [Update SpookyDB](https://github.com/FAForever/fa/actions/workflows/spookydb-update.yaml) workflow to update [SpookyDB](https://github.com/FAForever/spooky-db)
+- (4) Use the [Update UnitDB](https://github.com/FAForever/fa/actions/workflows/unitdb-update.yaml) workflow to update [UnitDB](https://github.com/FAForever/UnitDB)
```
[the remark this answers is not in the document]
Comment (Garanas) [reply docs/deployment.md:77]: It should happen automatically, but last time it did not.

File: docs/deployment.md (hunk 1 above)
Comment (Garanas) [reply docs/deployment.md]: Processed in 6cf8cca

**Pull request merged**